File: desafio1/src/Tickets/dto/update-ticket.dto.ts
import { IsString, IsOptional, IsNumber, Min, IsPositive } from 'class-validator';

export class UpdateTicketDto {
  @IsOptional()
  @IsString()
  name?: string;

  @IsOptional()
  @IsNumber()
  @IsPositive()
  @Min(1)
  price?: number;
}

File: flutter_desafio1/lib/features/events/models/update_ticket_dto.dart
class UpdateTicketDto {
  final String? name;
  final double? price;

  UpdateTicketDto({
    this.name,
    this.price,
  });

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    
    if (name != null) data['name'] = name;
    if (price != null) data['price'] = price;
    
    return data;
  }
}